src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -510,70 +511,94 @@ def handle_list_tasks_command(
     db_session: Session,
 ) -> None:
     """Handles the list tasks command."""
-    blocks = []
+    tasks = task_service.get_all_by_incident_id(
+        db_session=db_session,
+        incident_id=context["subject"].id,
+    )
 
     caller_only = False
+    user_id = None
     if body["command"] == context["config"].slack_command_list_my_tasks:
+        user_id = payload["user_id"]
         caller_only = True
 
+    if caller_only:
+        user_id = user_id
+        email = (client.users_info(user=user_id))["user"]["profile"]["email"]
+        user = user_service.get_or_create(
+            db_session=db_session,
+            organization=context["subject"].organization_slug,
+            user_in=UserRegister(email=email),
+        )
```

Comment:
Tasks are scoped to incidents and created and owned by participants. I think using `participant_service.get_by_incident_id_and_email()` would be more appropriate here if you want to check if an incident participant with this email address exists. If you don't need to perform this check, then you can just pass the resulting email from the `client.users_info()` call directly to `filter_tasks_by_assignee_and_creator()`

Reply:
Makes sense, if a task owner is always a participant, we can probably just pass the email from `client.users_info()` safely.